plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2729,7 +2729,7 @@ public boolean supportsReportingWrittenBytes(ConnectorSession session, SchemaTab
     private void cleanExtraOutputFiles(ConnectorSession session, String baseLocation, List<DataFileInfo> dataFiles)
     {
         Set<String> writtenFilePaths = dataFiles.stream()
-                .map(dataFileInfo -> baseLocation + "/" + dataFileInfo.getPath())
+                .map(dataFileInfo -> Location.of(baseLocation).appendPath(dataFileInfo.getPath()).toString())
                 .collect(toImmutableSet());
 
         cleanExtraOutputFiles(session, writtenFilePaths);
```

Comment:
Pass `Set<Location>` here.

You can make use of `io.trino.filesystem.Location#parentDirectory` in 
https://github.com/trinodb/trino/pull/17555/commits/e4b05c419b8ca5a23a741a3970740449073df265#diff-79f367e47040dda32ac81ecec46b27cfd7cc822858bc8c31fe435a15a47446dcR2744